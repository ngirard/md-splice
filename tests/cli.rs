use assert_cmd::Command;
use insta::assert_snapshot;

fn cmd() -> Command {
    Command::cargo_bin(env!("CARGO_PKG_NAME")).unwrap()
}

#[test]
fn test_i1_version_flag() {
    let output = cmd().arg("--version").output().unwrap();
    let stdout = String::from_utf8(output.stdout).unwrap();

    assert_snapshot!("i1_version", stdout, {
        ".version" => "md-splice [VERSION]"
    });
}

#[test]
fn test_i1_help_flag() {
    let output = cmd().arg("--help").output().unwrap();
    let stdout = String::from_utf8(output.stdout).unwrap();

    assert_snapshot!("i1_help", stdout, {
        ".version" => "md-splice [VERSION]"
    });
}

#[test]
fn test_i1_help_flag_insert() {
    let output = cmd().args(["insert", "--help"]).output().unwrap();
    let stdout = String::from_utf8(output.stdout).unwrap();

    assert_snapshot!("i1_help_insert", stdout, {
        ".version" => "md-splice [VERSION]"
    });
}

#[test]
fn test_i1_help_flag_replace() {
    let output = cmd().args(["replace", "--help"]).output().unwrap();
    let stdout = String::from_utf8(output.stdout).unwrap();

    assert_snapshot!("i1_help_replace", stdout, {
        ".version" => "md-splice [VERSION]"
    });
}
