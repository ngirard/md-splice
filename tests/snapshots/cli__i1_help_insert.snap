---
source: tests/cli.rs
expression: redact_version(&stdout)
---
Insert new Markdown content at a specified position

Usage: md-splice insert [OPTIONS]

Options:
  -c, --content <MARKDOWN_STRING>
          The Markdown content to insert or replace with

  -f, --file <FILE_PATH>
          The Markdown file to modify. [default: reads from stdin]

      --content-file <CONTENT_PATH>
          A file containing the Markdown content. Use '-' to read from stdin

  -o, --output <OUTPUT_PATH>
          Write the output to a new file instead of modifying the original

      --select-type <TYPE>
          Select node by type (e.g., 'p', 'h1', 'list', 'table')

      --select-contains <TEXT>
          Select node by its text content (fixed string)

      --select-regex <REGEX>
          Select node by its text content (regex pattern)

      --select-ordinal <N>
          Select the Nth matching node (1-indexed). Default is 1
          
          [default: 1]

      --after-select-type <TYPE>
          Restrict the search to the first match that occurs after another selector

      --after-select-contains <TEXT>
          Restrict the search to the first match that occurs after another selector

      --after-select-regex <REGEX>
          Restrict the search to the first match that occurs after another selector

      --after-select-ordinal <N>
          Choose the Nth landmark match for the `--after` selector (1-indexed)

      --within-select-type <TYPE>
          Restrict the search to nodes contained within another selector

      --within-select-contains <TEXT>
          Restrict the search to nodes contained within another selector

      --within-select-regex <REGEX>
          Restrict the search to nodes contained within another selector

      --within-select-ordinal <N>
          Choose the Nth landmark match for the `--within` selector (1-indexed)

      --until-type <TYPE>
          Select nodes up to (but not including) another selector

      --until-contains <TEXT>
          Select nodes up to (but not including) another selector

      --until-regex <REGEX>
          Select nodes up to (but not including) another selector

  -p, --position <POSITION>
          Position for the 'insert' operation

          Possible values:
          - before:        Insert before the selected node (as a sibling)
          - after:         Insert after the selected node (as a sibling)
          - prepend-child: Insert as the first child of the selected node/section
          - append-child:  Insert as the last child of the selected node/section
          
          [default: after]

  -h, --help
          Print help (see a summary with '-h')
